src/consumer.rs

```diff
@@ -137,6 +142,134 @@ impl PartitionConsumerJS {
 
         Ok(())
     }
+
+    #[node_bindgen(mt)]
+    async fn get_iterator(
+        &self,
+        offset: OffsetWrapper,
+    ) -> Result<PartitionConsumerIterator, FluvioError> {
+        if let Some(client) = &self.inner {
+            let stream = client.stream(offset.0).await?;
+            let mut foo = PartitionConsumerIterator::new();
+            foo.set_inner(Box::pin(stream));
+            Ok(foo)
+        } else {
+            Err(FluvioError::Other(CLIENT_NOT_FOUND_ERROR_MSG.to_owned()))
+        }
+    }
+}
+
+type PartitionConsumerIteratorInner = Pin<Box<dyn Stream<Item = Result<Record, FluvioError>>>>;
+
+pub struct PartitionConsumerIterator {
+    inner: Option<PartitionConsumerIteratorInner>,
+}
+
+#[node_bindgen]
+impl PartitionConsumerIterator {
+    #[node_bindgen(constructor)]
+    pub fn new() -> Self {
+        Self { inner: None }
+    }
+    pub fn set_inner(&mut self, client: PartitionConsumerIteratorInner) {
+        self.inner.replace(client);
+    }
+
+    #[node_bindgen]
+    fn next(&mut self) -> NextIter {
+        futures::executor::block_on(async {
+            if let Some(ref mut inner) = self.inner {
+                let next = inner.next().await;
+                let next = if let Some(next) = next {
+                    next
+                } else {
+                    return None.into()
+                };
+                let next = if let Ok(next) = next {
+                    next
+                } else {
+                    return None.into()
+                };
+                let next = if let Some(next) = next.try_into_bytes() {
+                    next
+                } else {
+                    return None.into()
+                };
+
+                if let Ok(msg) = String::from_utf8(next) {
+                    debug!("MSG: {:?}", msg);
+                    Some(msg).into()
+                } else {
+                    None.into()
+                }
+            } else {
+                None.into()
+            }
+        })
+    }
+}
+
+impl TryIntoJs for PartitionConsumerIterator {
+    fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
+        debug!("converting PartitionConsumerJS to js");
+        let new_instance = PartitionConsumerIterator::new_instance(js_env, vec![])?;
+        debug!("instance created");
+        if let Some(inner) = self.inner {
+            PartitionConsumerIterator::unwrap_mut(js_env, new_instance)?.set_inner(inner);
+        }
+        Ok(new_instance)
+    }
+}
+
+impl From<Option<String>> for NextIter {
+    fn from(value: Option<String>) -> Self {
+        let done = value.is_none();
+        Self {
+            value,
+            done,
+        }
+    }
+}
+
+#[derive(Debug)]
+struct NextIter {
```

Comment:
This should eventually get moved into `node-bindgen` and be a `NextIter<T: TryIntoJS>`